modules/frontend/mcp_tools.go

```diff
@@ -0,0 +1,361 @@
+package frontend
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/grafana/tempo/docs"
+	"github.com/grafana/tempo/pkg/api"
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/pkg/traceql"
+	"github.com/mark3labs/mcp-go/mcp"
+)
+
+const (
+	MetaTypeDocumentation   = "documentation"
+	MetaTypeSearchResults   = "search-results"
+	MetaTypeMetricsRange    = "metrics-range"
+	MetaTypeMetricsInstant  = "metrics-instant"
+	MetaTypeTrace           = "trace"
+	MetaTypeAttributeNames  = "attribute-names"
+	MetaTypeAttributeValues = "attribute-values"
+)
+
+func (s *MCPServer) handleTraceQLQuery(_ context.Context, _ mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+	level.Info(s.logger).Log("msg", "traceql query tool requested")
+
+	return toolResult(trimDocs(docs.TraceQLMain), MetaTypeDocumentation, "markdown", "1"), nil
+}
+
+func (s *MCPServer) handleTraceQLMetrics(_ context.Context, _ mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+	level.Info(s.logger).Log("msg", "traceql metrics tool requested")
+
+	return toolResult(trimDocs(docs.TraceQLMetrics), MetaTypeDocumentation, "markdown", "1"), nil
+}
+
+// handleSearch handles the traceql-search tool
+func (s *MCPServer) handleSearch(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
```

Comment:
should we log or capture the user-agent or some other metrics to know which Tempo API calls (search, metadata, etc) are coming from MCP and which ones are from grafana